etcdctl/command/util.go

```diff
@@ -213,9 +213,13 @@ func mustNewClient(c *cli.Context) client.Client {
 
 			// fail-back to try sync cluster with peer API. this is for making etcdctl work with etcd 0.4.x.
 			// TODO: remove this when we deprecate the support for etcd 0.4.
-			eps, serr := syncWithPeerAPI(c, ctx, hc.Endpoints())
-			if serr != nil {
-				handleError(ExitServerError, serr)
+			eps, trerr, serr := syncWithPeerAPI(c, ctx, hc.Endpoints())
+			if eps == nil {
+				if serr != nil {
```

Comment:
0.4 errors should not overwrite 2.0 ones. If 0.4 syncWithPeerAPI succeeds, then it is good. If it fails, we should parse the old error with the assumption that the cluster is a 2.0 one.